frontend/src/components/Contributions/ContributionListItem.vue

```diff
@@ -2,47 +2,100 @@
-        </div>
-        <div class="mr-2">
-          <span>{{ $t('contribution.date') }}</span>
-          <span>
-            {{ $d(new Date(contributionDate), 'monthAndYear') }}
-          </span>
-        </div>
-        <div class="mr-2">{{ memo }}</div>
-        <div v-if="type === 'pending' && !firstName" class="d-flex flex-row-reverse">
+          <div class="mr-2">
+            <span>{{ $t('contribution.date') }}</span>
+            <span>
+              {{ $d(new Date(contributionDate), 'monthAndYear') }}
+            </span>
+          </div>
+          <div class="mr-2">{{ memo }}</div>
           <div
-            class="pointer ml-5"
-            @click="
-              $emit('update-contribution-form', {
-                id: id,
-                contributionDate: contributionDate,
-                memo: memo,
-                amount: amount,
-              })
-            "
+            v-if="(state !== 'DELETED' && !firstName) || (state !== 'CONFIRMED' && !firstName)"
+            class="d-flex flex-row-reverse"
           >
-            <b-icon icon="pencil" class="h2"></b-icon>
-          </div>
-          <div class="pointer" @click="deleteContribution({ id })">
-            <b-icon icon="trash" class="h2"></b-icon>
+            <div
+              v-if="state !== 'CONFIRMED' && state !== 'DELETED'"
+              class="pointer ml-5"
+              @click="
+                $emit('update-contribution-form', {
+                  id: id,
+                  contributionDate: contributionDate,
+                  memo: memo,
+                  amount: amount,
+                })
+              "
+            >
+              <b-icon icon="pencil" class="h2"></b-icon>
+            </div>
+            <div
+              v-if="state !== 'CONFIRMED' && state !== 'DELETED'"
+              class="pointer"
+              @click="deleteContribution({ id })"
+            >
+              <b-icon icon="trash" class="h2"></b-icon>
+            </div>
+            <div v-if="messagesCount > 0" class="pointer">
+              <b-icon
+                v-b-toggle="collapsId"
+                icon="chat-dots"
+                class="h2 mr-5"
+                @click="getListContributionMessages"
+              ></b-icon>
+            </div>
           </div>
         </div>
+        <div v-if="messagesCount > 0">
+          <b-button
+            v-if="state === 'IN_PROGRESS'"
+            v-b-toggle="collapsId"
+            variant="warning"
+            @click="getListContributionMessages"
+          >
+            {{ $t('contribution.alert.answerQuestion') }}
+          </b-button>
+          <b-collapse :id="collapsId" class="mt-2">
+            <b-card>
+              <contribution-messages-list
+                :messages="messages_get"
+                :state="state"
+                :contributionId="contributionId"
+                @get-list-contribution-messages="getListContributionMessages"
+                @update-state="updateState"
```

Comment:
Please test for reemitation of this event like in all the other examples I gave you above.

Reply:
skiped